File: CSharp/day1.cs
//imports
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System;
using System.IO;

namespace ConsoleApp1
{
    static class day1
    {
        static readonly string[] inputs = File.ReadAllLines(Path.GetFullPath(Path.Combine(Directory.GetCurrentDirectory() + "..\\..\\..\\..\\inputs\\day1.txt")), Encoding.UTF8);

        //main
        static void Main(string[] args)
        {
            Part1();
        }

        static void Part1()
        {
            //How many measurements are larger than the previous measurement?

            float last_input = float.PositiveInfinity;
            int increased = 0;
            int decreased = 0;

            for (int i = 0; i < inputs.Length; i++)
            {
                if (int.Parse(inputs[i]) > last_input)
                {
                    increased += 1;
                }
                else
                {
                    decreased += 1;
                }
                last_input = int.Parse(inputs[i]);
            }

            Console.WriteLine(String.Format("How many measurements are larger than the previous measurement?\n{0}", increased));
        }

        static void Part2()
        {
            //How many sums are larger than the previous sum?
        }

    }
}

File: Python/day1.py
#imports
import os.path as Path

# set global path
path = Path.join(Path.dirname(__file__), 'inputs\\day1.txt')

#main
def part1():
    global path
    with open(path, 'r') as file:
        last_input = float("inf")
        increased = 0
        decreased = 0
        for item in file:
            item = int(item.strip())
            if item > last_input:
                increased += 1
            else:
                decreased += 1
            last_input = item
        print(f"\nHow many measurements are larger than the previous measurement?\n{increased}")

def part2():
    global path
    with open(path, 'r') as file:
        last_input = float("inf")
        increased = 0
        decreased = 0
        items = []
        for item in file:
            item = int(item.strip())
            items.append(item)
        for i in range(len(items) - 2):
            value = sum(items[i:i+3])
            if value > last_input:
                increased += 1
            else:
                decreased += 1
            last_input = value
        print(f"\nHow many sums are larger than the previous sum?\n{increased}")

def main():
    part1()
    part2()

if __name__ == "__main__":
    main()